src/test/java/pl/touk/sputnik/connector/http/SslVerificationTest.java

```diff
@@ -0,0 +1,168 @@
+        properties.setProperty(GeneralOption.HOST.getKey(), "localhost");
+        properties.setProperty(GeneralOption.PORT.getKey(), Integer.toString(untrustedService.httpsPort()));
+        properties.setProperty(GeneralOption.USE_HTTPS.getKey(), "true");
+        properties.setProperty(GeneralOption.VERIFY_SSL.getKey(), "false");
+        ConnectorDetails connectorDetails = new ConnectorDetails(ConfigurationBuilder.initFromProperties(properties));
+        HttpHelper httpHelper = new HttpHelper();
+        HttpHost httpHost = httpHelper.buildHttpHost(connectorDetails);
+        CloseableHttpClient closeableHttpClient = httpHelper.buildClient(httpHost, connectorDetails);
+
+        // when
+        CloseableHttpResponse response = closeableHttpClient.execute(httpHost, new HttpGet("/hello"));
+
+        // then
+        assertThat(response.getStatusLine().getStatusCode()).isEqualTo(200);
+    }
+
+    @Test
+    public void doNotVerifyHostnameWhenVerificationIsOff() throws Exception {
+        // given
+        Properties properties = new Properties();
+        properties.setProperty(GeneralOption.HOST.getKey(), "127.0.0.1");
+        properties.setProperty(GeneralOption.PORT.getKey(), Integer.toString(trustedService.httpsPort()));
+        properties.setProperty(GeneralOption.USE_HTTPS.getKey(), "true");
+        properties.setProperty(GeneralOption.VERIFY_SSL.getKey(), "false");
+        ConnectorDetails connectorDetails = new ConnectorDetails(ConfigurationBuilder.initFromProperties(properties));
+        HttpHelper httpHelper = new HttpHelper();
+        HttpHost httpHost = httpHelper.buildHttpHost(connectorDetails);
+        CloseableHttpClient closeableHttpClient = httpHelper.buildClient(httpHost, connectorDetails);
+
+        // when
+        CloseableHttpResponse response = closeableHttpClient.execute(httpHost, new HttpGet("/hello"));
+
+        // then
+        assertThat(response.getStatusLine().getStatusCode()).isEqualTo(200);
+    }
+
+    @Test
+    public void verifySslTrust() throws Exception {
+        // given
+        System.setProperty("javax.net.ssl.trustStore", this.getClass().getResource("/ssl/trusted/cacerts.jks").getPath());
+        System.setProperty("javax.net.ssl.trustStorePassword", "changeit");
+        Properties properties = new Properties();
+        properties.setProperty(GeneralOption.HOST.getKey(), "localhost");
+        properties.setProperty(GeneralOption.PORT.getKey(), Integer.toString(trustedService.httpsPort()));
+        properties.setProperty(GeneralOption.USE_HTTPS.getKey(), "true");
+        properties.setProperty(GeneralOption.VERIFY_SSL.getKey(), "true");
+        ConnectorDetails connectorDetails = new ConnectorDetails(ConfigurationBuilder.initFromProperties(properties));
+        HttpHelper httpHelper = new HttpHelper();
+        HttpHost httpHost = httpHelper.buildHttpHost(connectorDetails);
+        CloseableHttpClient closeableHttpClient = httpHelper.buildClient(httpHost, connectorDetails);
+
```

Comment:
Maybe first 7 lines can be extracted? You need constructed objects, but Properties section can be extracted to a method.